src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js

```diff
@@ -227,7 +227,9 @@ const filterCells = (cellSets, sampleKey, groupBy) => {
   let cellSetLegendsData = [];
   const addedCellSetKeys = new Set();
 
-  filteredCells = filteredCells.map((cell) => {
+  filteredCells = filteredCells.reduce((acc, cell) => {
```

Comment:
From what I'm getting this wasn't part of the changes, it was a performance improvement, right?

I was thinking, using a `reduce` vs a `map` + `filter` is less declarative. I feel like perhaops we shudl consider using map and filter when this doesn't improve the performance much.

I was reading [this](https://www.davideasaf.com/mapfilter-vs-reduce/) a little bit which I think fits this case.

Reply:
A `map` + `filter` produces arrays, whereas in this case I'm making a dictionary object with the `cellId` as the property, which would make it much easier and faster to access the properties of a cell later on down the line.

This reduce can be replaced with a `forEach` which adds to an empty object.

Reply:
Ah ok, never mind then.

About using a `forEach` or a `reduce` I think both options are fine, if you want to keep it like this I understand now and it makes sense